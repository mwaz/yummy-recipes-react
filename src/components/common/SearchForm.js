import React from 'react';

import { Form, Row, Col, Button } from 'react-bootstrap';


/**
 * Component to handle search for both categeries and recipes
 */

const SearchForm = (props) => {
  return (
    <Form onSubmit={props.search_event}>
      <Row style={{ float: 'right' }} >
        <Col sm={8} >
          <input name="search" className="form-control" placeholder={props.search_placeholder} onKeyDown={props.name_change} onKeyPress={props.name_change} onChange={props.search} />
        </Col>
        <Col sm={3}>
          <Button id="mysearch" bsStyle="info" type="submit" onClick={props.clear}> Clear Search </Button>
        </Col>
      </Row>
    </Form>
  );
};
export default SearchForm;
