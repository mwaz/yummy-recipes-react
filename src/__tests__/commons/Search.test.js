import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import SearchForm from '../../components/common/SearchForm';

describe('SearchForm component', () => {
  const search = jest.fn();
  const wrapper = shallow(<SearchForm search={search} location={{}} />);

  it('renders properly without crashing', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('has clickable button', () => {
    expect(wrapper.find('Button').length).toBe(1);
    expect(wrapper.find('Button').simulate('click'));
  });
});

