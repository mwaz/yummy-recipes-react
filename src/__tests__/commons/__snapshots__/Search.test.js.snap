// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component renders properly without crashing 1`] = `
<Form
  bsClass="form"
  componentClass="form"
  horizontal={false}
  inline={false}
>
  <Row
    bsClass="row"
    componentClass="div"
    style={
      Object {
        "float": "right",
      }
    }
  >
    <Col
      bsClass="col"
      componentClass="div"
      sm={8}
    >
      <input
        className="form-control"
        name="search"
        onChange={[Function]}
      />
    </Col>
    <Col
      bsClass="col"
      componentClass="div"
      sm={3}
    >
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="info"
        disabled={false}
        id="mysearch"
        type="submit"
      >
         Clear Search 
      </Button>
    </Col>
  </Row>
</Form>
`;
